src/main/java/org/keycloak/plugins/groups/services/VoAdminGroup.java

```diff
@@ -0,0 +1,103 @@
+import org.keycloak.plugins.groups.representations.GroupConfigurationRepresentation;
+import org.keycloak.services.ServicesLogger;
+import org.keycloak.services.resources.admin.permissions.AdminPermissionEvaluator;
+import org.keycloak.theme.FreeMarkerUtil;
+
+public class VoAdminGroup {
+        private final KeycloakSession session;
+        private final RealmModel realm;
+        private final UserModel voAdmin;
+        private GroupModel group;
+        private final GroupConfigurationRepository groupConfigurationRepository;
+        private final UserVoGroupMembershipRepository userVoGroupMembershipRepository;
+        private final CustomFreeMarkerEmailTemplateProvider customFreeMarkerEmailTemplateProvider;
+
+        public VoAdminGroup(KeycloakSession session, RealmModel realm, UserModel voAdmin, GroupConfigurationRepository groupConfigurationRepository, UserVoGroupMembershipRepository userVoGroupMembershipRepository, GroupModel group) {
+            this.session = session;
+            this.realm =  realm;
+            this.voAdmin = voAdmin;
+            this.group = group;
+            this.groupConfigurationRepository =  groupConfigurationRepository;
+            this.userVoGroupMembershipRepository =  userVoGroupMembershipRepository;
+            this.customFreeMarkerEmailTemplateProvider = new CustomFreeMarkerEmailTemplateProvider(session, new FreeMarkerUtil());
+            this.customFreeMarkerEmailTemplateProvider.setRealm(realm);
+        }
+
+        @GET
+        @Produces("application/json")
+        public GroupConfigurationRepresentation getGroupConfiguration() {
+            GroupConfigurationEntity groupConfiguration = groupConfigurationRepository.getEntity(group.getId());
+            //if not exist, group have only created from main Keycloak
+            if(groupConfiguration == null) {
+                return new GroupConfigurationRepresentation(group.getId());
+            } else {
+                return EntityToRepresentation.toRepresentation(groupConfiguration, realm);
+            }
+        }
+
+        @POST
+        @Consumes(MediaType.APPLICATION_JSON)
+        public Response saveGroupConfiguration(GroupConfigurationRepresentation rep) {
+            GroupConfigurationEntity entity = groupConfigurationRepository.getEntity(group.getId());
+            if ( entity != null) {
+                groupConfigurationRepository.update(entity, rep, voAdmin.getId());
+            } else {
+                //only group exists
+                groupConfigurationRepository.create(rep, group.getId(), voAdmin.getId());
+            }
+            //aup change action
+            return Response.noContent().build();
+        }
+
+        @POST
+        @Path("/member/{userId}/admin")
+        @Produces("application/json")
+        public Response addOrRemoveVoAdmin(@PathParam("userId") String userId, @QueryParam("addVoMember") Boolean addVoMember) {
+            UserModel user = session.users().getUserById(realm, userId);
```

Comment:
Shouldn't this be restricted only to users with the correct permissions?
Only the realm admin should be allowed to perform this operation, correct?

Reply:
ok, i see you have the check on the VoAdminService. :)
You don't have on the @GET, but i assume that it will not disclose any information (returns always empty) to users who are not voadmins.

Reply:
In order to call any VoAdminService web service you need to be VO admin of this group.
I do not add any request method (fe @GET) because VoAdminService methods decide about request methods.

Reply:
yes, i was referring to `VoAdminService.getVoAdminGroups()` which is open to any user. Anyway, unauthorized (regular users) will just get empty responses.